### PR TITLE
LM-50: modify services and controllers to handle new project model

### PR DESCRIPTION
**Pull request by rgillen-uno**
Fixing the services, controllers and tests that were broken by the Model changes introduced by [LM-13-1](https://github.com/UnosquareBelfast/admin-backend/pull/82).

### CODE REVIEW

File: AdminCore.Services.Tests/DashboardServiceLogicTests.cs [hunk 1]
```diff
@@ -18,10 +18,10 @@ public DashboardServiceLogicTests()
     [Fact]
     public void TestEmployeeDashboardEventsLogic_Check1ResultComesBackForEmployeeId1ForDecember()
     {
-      // Employee 1 has 2 December 2018 events. One approved and one cancelled. This query should only return the one that is approved.
+      // Employee 1 has 2 December 2018 events. One Awaiting approval and one cancelled. This query should only return the one that is awaiting approved.
       var resultList = _mockDatabase.EventRepository.Where(evnt => DashboardService.EmployeeDashboardEventsQuery(1, new DateTime(2018, 12, 5), evnt)).ToList();
       Assert.Single(resultList);
-      Assert.Equal((int)EventStatuses.Approved, resultList.First().EventStatusId);
+      Assert.Equal((int)EventStatuses.AwaitingApproval, resultList.First().EventStatusId);
```
Comment (kkuczkowski-unosq) [inline AdminCore.Services.Tests/DashboardServiceLogicTests.cs:37]: reason for this change is unclear, why was this changed?

File: AdminCore.Services.Tests/MappingProfileTests/ContractMappingProfileTests.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+{
+    public class ContractMappingProfileTests
+    {
+        private IMapper _mapper;
+
+        public ContractMappingProfileTests()
+        {
+            _mapper = new Mapper(new MapperConfiguration(cfg =>
+            {
+//                cfg.ValidateInlineMaps = false;
```
Comment (kkuczkowski-unosq) [inline AdminCore.Services.Tests/MappingProfileTests/ContractMappingProfileTests.cs]: clean up?

**Review by kkuczkowski-unosq — APPROVED** [review]
looks good to me :shipit:

File: AdminCore.Services.Tests/DashboardServiceLogicTests.cs (hunk 1 above)
Comment (rgillen-uno) [reply AdminCore.Services.Tests/DashboardServiceLogicTests.cs:37]: The eventRepo didn't contain an event that was Approved with these particular parameters, so the test was failing. So I changed the assertion to assert for an event that did exist in the mock db.

File: AdminCore.Services.Tests/MappingProfileTests/ContractMappingProfileTests.cs (hunk 2 above)
Comment (rgillen-uno) [reply AdminCore.Services.Tests/MappingProfileTests/ContractMappingProfileTests.cs]: Cleaned comments.

File: AdminCore.DatabaseMigration/db/migrations/V1.11__add_project.sql [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+-- Set the project_id column on team to the project id that matches the old team => client relationship.
+-- Ensure that there is a default project created for every existing team/client and the foreign keys are assigned correctly before dropping data.
+UPDATE public.team
+SET
+  project_id = project.project_id
+FROM
+  project
+WHERE
+  team.client_id = project.client_id;
+
```
Comment (BrownMark) [inline AdminCore.DatabaseMigration/db/migrations/V1.11__add_project.sql]: Make project_id not nullable.

**Review by BrownMark — CHANGES_REQUESTED** [review]

File: AdminCore.DatabaseMigration/db/migrations/V1.11__add_project.sql (hunk 3 above)
Comment (rgillen-uno) [reply AdminCore.DatabaseMigration/db/migrations/V1.11__add_project.sql]: Done
